tests/Kernel/MediaRenderTest.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_theme\Kernel;
+
+use Drupal\media\Entity\Media;
+use Symfony\Component\DomCrawler\Crawler;
+
+/**
+ * Tests that our media types render with correct markup.
+ */
+class MediaRenderTest extends AbstractKernelTestBase {
+
+  /**
+   * The media storage.
+   *
+   * @var Drupal\media\MediaStorage
+   */
+  protected $mediaStorage;
+
+  /**
+   * The media view builder.
+   *
+   * @var \Drupal\Core\Entity\EntityViewBuilderInterface
+   */
+  protected $mediaViewBuilder;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'field',
+    'file',
+    'media',
+    'oe_media',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->installSchema('file', 'file_usage');
+    $this->installEntitySchema('media');
+    $this->installEntitySchema('file');
+
+    $this->installConfig([
+      'file',
+      'media',
+      'oe_media',
+    ]);
+
+    $this->mediaStorage = $this->container->get('entity_type.manager')->getStorage('media');
+    $this->mediaViewBuilder = $this->container->get('entity_type.manager')->getViewBuilder('media');
+  }
+
+  /**
+   * Tests that the Document media type is rendered with the correct ECL markup.
+   */
+  public function testDocumentMedia(): void {
+    $file = file_save_data(file_get_contents(drupal_get_path('module', 'oe_media') . '/tests/fixtures/sample.pdf'), 'public://test.pdf');
```

Comment:
Shouldn't we test this with private file system?

Reply:
Unfortunatly it is not easy. Can't use private file system inside KernelTests.
I tried but didn't find solution.